exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java

```diff
@@ -205,38 +212,97 @@ void parseServicesResponse() {
     assertThat(actual, contains(expected.toArray()));
   }
 
-  private static Stream<Arguments> testData() {
+  private static Collection<Arguments> txResponseTestData() {
+    List<Arguments> arguments = new ArrayList<>();
+
+    // Add success status
     String successStatus = "'status': {\n"
         + "    'type': 'success'\n"
         + "}\n";
+    arguments.add(arguments(ExecutionStatuses.SUCCESS, successStatus));
 
+    // Add error statuses of various types from various call site types:
     int errorCode = 1;
     String errorDescription = "Some error";
+    int runtimeId = 2;
+    int instanceId = 3;
+    int methodId = 4;
+    String interfaceId = "exonum.Configure";
+
+    ExecutionError errorTemplate = ExecutionError.newBuilder()
+        .setCode(errorCode)
+        .setDescription(errorDescription)
+        .setRuntimeId(runtimeId)
+        .buildPartial();
+    CallSite callSiteTemplate = CallSite.newBuilder()
+        .setInstanceId(instanceId)
+        .setMethodId(methodId)
+        .setInterface(interfaceId)
+        .buildPartial();
+
     String errorStatusTemplate = "'status': {\n"
         + "    'type': '%s',\n"
         + "    'code': " + errorCode + ",\n"
-        + "    'description': \"" + errorDescription + "\""
+        + "    'description': \"" + errorDescription + "\","
+        + "    \"runtime_id\": " + runtimeId + ",\n"
+        + "    \"call_site\": {\n"
+        + "            \"instance_id\": " + instanceId + ",\n"
+        + "            \"call_type\": \"%s\",\n"
+        + "            \"method_id\": " + methodId + ",\n"
+        + "            \"interface\": \"" + interfaceId + "\"\n"
+        + "    }\n"
         + "}\n";
 
-    String serviceErrorStatus = String.format(errorStatusTemplate, "service_error");
-    String dispatcherErrorStatus = String.format(errorStatusTemplate, "dispatcher_error");
-    String runtimeErrorStatus = String.format(errorStatusTemplate, "runtime_error");
-
-    String panicStatus = "'status': {\n"
-        + "    'type': 'panic',\n"
-        + "    'description': \"" + errorDescription + "\""
-        + "}\n";
+    @Value
+    class StatusParameters {
+      ErrorKind errorKind;
+      String errorKindStr;
+      CallSite.Type callSiteType;
+      String callSiteTypeStr;
+    }
 
-    return Stream.of(
-        arguments(ExecutionStatuses.success(), successStatus),
-        arguments(ExecutionStatuses.serviceError(errorCode, errorDescription), serviceErrorStatus),
-        arguments(ExplorerApiHelper.buildExecutionStatus(ErrorKind.DISPATCHER, errorCode,
-            errorDescription),
-            dispatcherErrorStatus),
-        arguments(ExplorerApiHelper.buildExecutionStatus(ErrorKind.RUNTIME, errorCode,
-            errorDescription),
-            runtimeErrorStatus),
-        arguments(ExplorerApiHelper.buildPanicExecutionStatus(errorDescription), panicStatus)
+    List<StatusParameters> combinations = ImmutableList.of(
```

Comment:
Alternatively, we could test _separately_:
* All the callSite types.
* All the status types (with same call site).